### ISSUE TITLE
The training with Model Builder is utterly slow

### ISSUE THREAD

**Issue by forReason** [dotnet/machinelearning#5392]
### System information

- Win 10, 1909
- .net core 3.1.402
- Intel core I7 3770k
- Nvidia RTX 2080

### Issue

- **What did you do?**
1. I created an image dataset similar to mnist, roughly 10000 images, each image 64x36 px, 2 classes
[dataset (32mb)](http://julianbechtold.de/media/AI.zip)
2. Opened the model builder (first time to try ml.net), found no method to create your own model, so i choose image classification.
3. Started training

- **What happened?**
After > 1h of training and seeing in the log that each iteration of one image(!!!) takes roughly 0.5 seconds and it was still at the step of "Bottleneck computation", I stopped training
![image](https://user-images.githubusercontent.com/12736950/92920802-2f29cf00-f433-11ea-939f-1ed82be1412e.png)

Additionally I saw that a way oversized network with transfer learning was thrown at the task whereas I manually would have assigned 1-2 Convolutional layers (if even) + a couple of normal connected layers.

- **What did you expect?**
I expected training to take max 10 Minutes.
If I approached mnist with tensorflow, Id be going with a simple, ready model in roughly 10 Minutes. The main purpose of ml.net for me was the compatibility with c# but I was quite unimpressed to see the capabilities of the Model builder if I can only select 4 preset scenarios which I cannot even customize to my needs.

### Source code / logs
No soucecode so far, as I dont get beyond the training.

**Comment by forReason** [comment]
Update > 3h later.... still training, for a small mnist.

**Comment by antoniovs1029** [comment]
Hi, @forReason . I think several of the points you're touching are particular to ModelBuilder, which is a VS plugin that uses ML.NET behind the scenes. The repo to post issues and suggestions on ModelBuilder is this one:
https://github.com/dotnet/machinelearning-modelbuilder

I'm tagging @LittleLittleCloud so that the ModelBuilder team can take a look into this. Among the ModelBuilder comments you left I can see that you're requesting: Ability to set the amount of time ModelBuilder runs, ability to set a desired accuracy after which it's ok to stop the model training, ability to use GPU, more clarity about how far ModelBuilder is in the process (i.e. stating more clearly how many models have been tried, and how many are left) (particular for Image Classification), and ability to cancel without loosing all of the already trained models.

You also want more control on the models that are going to be used. For this, I think it would be harder for ModelBuilder, since it's meant more like a general AutoML tool that facilitates automatically training preset models. So for this, I'd recommend that you actually use ML.NET directly, as that would give you more control over the parameters you want to use.

Also, please notice that ML.NET isn't a deep learning tool, so you won't be able to create Convolutional networks with it. Still, ML.NET does offer the capability to load Onnx and Tensorflow pretrained models to run inferencing. It seems this would appeal to you as you mentioned you're particularly interested in ML.NET/ModelBuilder because of the C# support. To see how to load pre-trained Onnx and Tensorflow models please refer to these samples:

https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/DeepLearning_ImageClassification_TensorFlow

https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/DeepLearning_ObjectDetection_Onnx

Regarding your model training being very slow: I'm not very familiar with actually using the [ImageClassificationTrainer](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.vision.imageclassificationtrainer?view=ml-dotnet), so I wouldn't know if the time it takes to train your model is expected or not (which would also depend on your machine, and whatever parameters ModelBuilder is using behind the scenes). So, if using pretrained TF or Onnx models is not an option, I'd recommend you actually using the `ImageClassificationTrainer` on your own, to see if it still takes a lot of time to train, even with your desired parameters.

The list of options for ImageClassificationTrainer are here (Notice the options related to Bottleneck, which might be related to some of the comments you've made, the EarlyStopiingCriteria related to stop training depending on the accuracy, and :
https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.vision.imageclassificationtrainer.options?view=ml-dotnet

To create an ImageClassificationTrainer we have these APIs:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.visioncatalog.imageclassification?view=ml-dotnet#Microsoft_ML_VisionCatalog_ImageClassification_Microsoft_ML_MulticlassClassificationCatalog_MulticlassClassificationTrainers_System_String_System_String_System_String_System_String_Microsoft_ML_IDataView_

To actually use GPU you'd only need to install the dependencies listed here (notice that ModelBuilder won't use GPU even if you install these... you'd need to use ML.NET directly and install these to use GPU):
https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.vision.imageclassificationtrainer?view=ml-dotnet

**Comment by harishsk** [comment]
Hi @forReason, I am transferring the issue to ModelBuilder to see if they can address some of our concerns.

@LittleLittleCloud, Please transfer it back with your recommendations if you think more work needs to be done on ML.NET.
